src/test/kotlin/org/rust/lang/core/resolve/RsStdlibResolveTest.kt

```diff
@@ -254,6 +254,30 @@ class RsStdlibResolveTest : RsResolveTestBase() {
     """)
     }
 
+    fun `test inherent impl array coercing to slice`() = stubOnlyResolve("""
+    //- main.rs
+        fn main() { [1u8, 2u8, 3u8].len(); }
```

Comment:
All the tests are depends on a single std method len(), so it is reasonable to move they into `RsGenericExpressionTypeInferenceTest` and implement slice.len() manually

Reply:
Is placing it in `RsTypeAwareResolveTest` also OK?
There is already a similar `test slice resolve` there.

Reply:
All current tests passes without your fix, so they are not enough "similiar" :) So place it next to the exists one.

Reply:
Hmm, really there are some mirroring between our isolated tests and std-base. But non-std-based unit-like tests should be preferred because they are faster and more clear & visible. but `RsGenericExpressionTypeInferenceTest` is really wrong here, we are speaking about method resolution but not type inference. So I think we should use `RsTypeAwareResolveTest`, there are already some `slice resolve` tests